### PR TITLE
Add release-1.5 branch config for CAPM3 and IPAM

### PR DESCRIPTION
**Pull request by smoshiur1237**
release-1.5 branch configuration for CAPM3 and IPAM

### CODE REVIEW

File: lib/releases.sh [hunk 1]
```diff
@@ -70,20 +70,17 @@ CAPM3RELEASEPATH="{https://api.github.com/repos/${CAPM3_BASE_URL:-metal3-io/clus
...
   export CAPM3RELEASE="${CAPM3RELEASE:-$(get_latest_release "${CAPM3RELEASEPATH}" "v1.4.")}"
   export CAPIRELEASE="${CAPIRELEASE:-$(get_latest_release "${CAPIRELEASEPATH}" "v1.4.")}"
+elif [ "${CAPM3RELEASEBRANCH}" = "release-1.5" ]; then
+  export CAPM3RELEASE="${CAPM3RELEASE:-$(get_latest_release "${CAPM3RELEASEPATH}" "v1.5.")}"
+  export CAPIRELEASE="${CAPIRELEASE:-$(get_latest_release "${CAPIRELEASEPATH}" "v1.5.")}"
 else
-  # We need to fix this to a non-existent CAPM3 release version to make sure
-  # the local override created for main branch is not conflicting with existing
-  # release tag. 
-  export CAPM3RELEASE="v1.5.99"
```
[the remark this answers is not in the document]
Comment (smoshiur1237) [reply lib/releases.sh:86]: I am keeping it as 1.5.99, bump will be needed when we have release-1.6.

[the remark this answers is not in the document]
Comment (Sunnatillo) [reply lib/releases.sh:86]: AFAIK  this should be fine. v1.5.99 used to create local override for main branch. We set it to 99 to not to conflict with existing release tag.

**Review by Sunnatillo — APPROVED** [review]
/lgtm

File: lib/releases.sh [hunk 2]
```diff
@@ -70,19 +70,19 @@ CAPM3RELEASEPATH="{https://api.github.com/repos/${CAPM3_BASE_URL:-metal3-io/clus
...
   export CAPIRELEASE="${CAPIRELEASE:-$(get_latest_release "${CAPIRELEASEPATH}" "v1.4.")}"
+elif [ "${CAPM3RELEASEBRANCH}" = "release-1.5" ]; then
+  export CAPM3RELEASE="${CAPM3RELEASE:-$(get_latest_release "${CAPM3RELEASEPATH}" "v1.5.")}"
+  export CAPIRELEASE="${CAPIRELEASE:-$(get_latest_release "${CAPIRELEASEPATH}" "v1.5.")}"
 else
   # We need to fix this to a non-existent CAPM3 release version to make sure
   # the local override created for main branch is not conflicting with existing
-  # release tag. 
+  # release tag.
   export CAPM3RELEASE="v1.5.99"
```
Comment (tuminoid) [inline lib/releases.sh:86]:
```suggestion
  export CAPM3RELEASE="v1.6.99"
```
It should be bumped as well like @lentzi90 said.

Comment (tuminoid) [reply lib/releases.sh:86]: Nevermind

Comment (kashifest) [reply lib/releases.sh:86]: No I guess, 1.5.99 is the correct version for now, we would change it to 1.5.0 one we have proper release

Comment (kashifest) [reply lib/releases.sh:86]: just to be clear once we have the branch, we would have a check for release-branch-0.5 which would point to v1.5.0 and then we can proceed with 1.6.99 for main branch

**Review by Sunnatillo — APPROVED** [review]
/lgtm

**Comment by Sunnatillo** [comment]
/hold 
Until we discuss when we  `next release` to `v1.6.99`

**Comment by lentzi90** [comment]
Adding for reference how CAPI is configuring things.
In their e2e tests they use v1.6.99: https://github.com/kubernetes-sigs/cluster-api/blob/abab5280a2c4cfd26b385bc09ef69eeaa11f8e56/test/e2e/config/docker.yaml#L65
And then they have metadata like this: https://github.com/kubernetes-sigs/cluster-api/blob/main/test/e2e/data/shared/main/metadata.yaml

**Comment by kashifest** [comment]
> Adding for reference how CAPI is configuring things. In their e2e tests they use v1.6.99: https://github.com/kubernetes-sigs/cluster-api/blob/abab5280a2c4cfd26b385bc09ef69eeaa11f8e56/test/e2e/config/docker.yaml#L65 And then they have metadata like this: https://github.com/kubernetes-sigs/cluster-api/blob/main/test/e2e/data/shared/main/metadata.yaml

Yes, without adding the metadata , just adding v1.6.99 wouldn't work. And that would also mean our main branch would start implementing contract changes if any towards CAPI v1.6

**Comment by Sunnatillo** [comment]
/unhold 
We decided that the changes should land after branch out

**Review by lentzi90 — COMMENTED** [review]
/approve

**Pull request merged**